src/components/engine/engine-instances-table.tsx

```diff
@@ -276,44 +288,30 @@ const EditModal = ({
   refetch: () => void;
 }) => {
   const toast = useToast();
+  const { mutate: editInstance } = useEngineEditInstance();
   const { onClose } = disclosure;
 
-  const form = useForm({
+  const form = useForm<EditEngineInstanceInput>({
     defaultValues: {
-      name: instance?.name,
-      url: instance?.url,
+      instanceId: instance.id,
+      name: instance.name,
+      url: instance.url,
     },
   });
 
-  const onSubmit = async (data: { name: string; url: string }) => {
-    // Instance URLs should end with a /.
-    const url = data.url.endsWith("/") ? data.url : `${data.url}/`;
-
+  const onSubmit = async (data: EditEngineInstanceInput) => {
     try {
-      const res = await fetch(`${THIRDWEB_API_HOST}/v1/engine/${instance.id}`, {
-        method: "PUT",
-        credentials: "include",
-        headers: {
-          "Content-type": "application/json",
-        },
-        body: JSON.stringify({
-          name: data.name,
-          url,
-        }),
-      });
-      if (!res.ok) {
-        throw new Error(`Unexpected status ${res.status}`);
-      }
+      await editInstance(data);
       toast({
         status: "success",
-        description: "Updated an Engine instance.",
+        description: "Successfully updated this Engine.",
       });
       refetch();
       onClose();
     } catch (e) {
       toast({
         status: "error",
-        description: "Error updating an Engine instance.",
+        description: "Error updating this Engine.",
       });
     }
```

Comment:
if you expose the `mutate` function already then just use the `mutate(<data>, {onSuccess, onError})` callback here -> that way you don't double implement all of this stuff with try catch here, move the toast and onClose to `onSuccess` callback of mutate, call mutate directly in the `onSubmit` case, same with `onError`